### DISCUSSION TITLE
IdentityPartsSeparator is ignored

### DISCUSSION THREAD

**Discussion by pahlgrimm** [ravendb/ravendb#13032] [Q&A] [answered]
Hi I want wo create Id like "Folder/" and my IdentityPartsSeparator should be '_'. But when I store my objects and .SaveChanges() the Id is "Folder/0....1-A" instead of "Folder_0...1-A"
What am I missing? 

My Code:

using (IDocumentStore store = new DocumentStore
{
    Urls = new[]
    {
        "http://ip:8888"
    },
    Database = "DBase",
    Conventions =
    {
        
        IdentityPartsSeparator = '_'
    }
}.Initialize())
{

    IDocumentSession session = store.OpenSession();

    var stacks = session.Query<Folder>().ToList();
    if (stacks.Count == 0)
    {
        var firstStack = new Folder
        {
            Id = "Folder/",
            Name = "Name",
            ChangedAt = DateTime.Now,
            CreatedAt = DateTime.Now,
            CreatedBy = "Mops"
        };

        session.Store(firstStack);
    }
    session.SaveChanges();

**Comment by haludi** [comment, marked as answer]
Hi,

By using id with slash in the end you are using server side id generation.
This convention is for hilo.
Just don’t set the id.


[Fact]
public async Task TestCase()
{
    using var store = GetDocumentStore(new Options {
ModifyDocumentStore = s => s.Conventions.IdentityPartsSeparator = "_"
});
    var session = store.OpenSession();
    var firstStack = new Folder
    {
        Name = "Name",
        ChangedAt = DateTime.Now,
        CreatedAt = DateTime.Now,
        CreatedBy = "Mops"
    };

    session.Store(firstStack);
    session.SaveChanges();
}

Best regards,
Igal

**Reply by ayende** [reply]
You can also configure the separator on the server side here:
![image](https://user-images.githubusercontent.com/116915/141674181-1fc04b36-48cb-4371-a3e0-3d0d14079698.png)

**Comment by pahlgrimm** [comment]
Okay, thanks for your fast answers! Then I let the ids be.

Tanks alot!

Patrick
